src/noit_check_lmdb.c

```diff
@@ -1069,67 +1069,126 @@ noit_check_lmdb_remove_check_from_db(uuid_t checkid, mtev_boolean force) {
   return rc;
 }
 
-int
-noit_check_lmdb_delete_check(mtev_http_rest_closure_t *restc,
-                             int npats, char **pats) {
-  mtev_http_session_ctx *ctx = restc->http_ctx;
-  uuid_t checkid;
-  noit_check_t *check = NULL;
-  int rc, error_code = 500;
-  noit_lmdb_instance_t *instance = noit_check_get_lmdb_instance();
-  mtevAssert(instance != NULL);
+typedef struct lmdb_delete_check_closure {
+  mtev_http_rest_closure_t *restc;
+  uuid_t check_uuid;
+  int error_code;
+} lmdb_delete_check_closure_t;
 
-  if(npats != 2) goto error;
-  if(mtev_uuid_parse(pats[1], checkid)) goto error;
+static void
+noit_check_lmdb_delete_check_free_closure(void *closure) {
+  lmdb_delete_check_closure_t *ldcc = (lmdb_delete_check_closure_t *)closure;
+  if (ldcc) {
+    free(ldcc);
+  }
```

Comment:
I feel confused by this. First, I didn't think a C cast could fail like this, and second doesn't `free` take a `void *` anyway?

Reply:
I guess the thing that threw me off, then, is that, at least according to the docs, `free` can safely take `NULL` as a parameter. Doesn't matter to me either way, but it seemed odd.

Reply:
An example of how this typically works can be found here:

https://github.com/circonus-labs/reconnoiter/blob/master/src/noit_check_lmdb.c#L618

Here, the struct has internal members requiring manual free. Hence, the need to cast. Despite the fact that there are no internal members that need freeing at the moment, if we add some at a later date, we will need the cast. The cast is extremely cheap and documents the type, since mtev eventer callbacks take a void*. This cast lets anyone trying to modify the code what the actual type is.

Reply:
``free`` does accept null, the ``if`` statement was not required.  I usually don't like adding code for nothing but in this case it's very likely that more fields could be added in the future.  Can call ``free`` on it but can derefence any of its future members